### PR TITLE
fix: show short url for cloud

### LINKED ISSUE #4295: [FEATURE]: Show a shortlink for cloud project URL instead of long links
## Feature Request
When a user connects to cloud using `garden login`, the cloud project link that's provided to user is quite long at the moment and doesn't fit the cli aesthetics. Currently the link looks like this

```
https://app.garden.io/projects/4fc1331d-7742-43a4-9f16-acf90481c68a?sessionId=4fc1331d-7742-43a4-9f16-acf90481c68a&userId=4fc1331d-7742-43a4-9f16-acf90481c68a
```

Would be great aesthetically if a shortlink is provided instead of massive url.

### Background / Motivation

The massive url doesn't look nice, and goes on multiple lines on small window cli atm. A shortlink instead could be much smaller and would fit better aesthetically.

### Suggested Implementation(s)

Once session is registered to cloud, core should receive a short link in response from cloud that it shows to the user. The short link could look like: 

```
https://app.garden.io/go/9w5j
```

### How important is this feature for you/your team?

🌵 Not having this feature makes using Garden painful

### PR DESCRIPTION
**Pull request by shumailxyz**
**What this PR does / why we need it**:
Instead of current lengthy urls in cli, it shows a short url of  format <domain>/go/<shortlink>.

**Which issue(s) this PR fixes**:

Fixes #4295

### CODE REVIEW

File: core/src/commands/base.ts [hunk 1]
```diff
@@ -225,6 +246,7 @@ export abstract class Command<A extends Parameters = {}, O extends Parameters =
       const distroName = getCloudDistributionName(cloudSession.api.domain)
       const userId = (await cloudSession.api.getProfile()).id
       const commandResultUrl = cloudSession.api.getCommandResultUrl({
+        shortLink: cloudSession.shortLink,
```
Comment (shumailxyz) [inline core/src/commands/base.ts:249]: This is the only change here. Rest of the changes is prettier being prettier.

File: core/src/commands/base.ts [hunk 2]
```diff
@@ -269,13 +290,14 @@ export abstract class Command<A extends Parameters = {}, O extends Parameters =
       const distroName = getCloudDistributionName(cloudSession.api.domain)
       const userId = (await cloudSession.api.getProfile()).id
       const commandResultUrl = cloudSession.api.getCommandResultUrl({
+        shortId: cloudSession.shortId,
```
Comment (shumailxyz) [inline core/src/commands/base.ts:293]: This is the only important change here. Rest of the stuff is prettier being prettier.

File: core/src/cloud/api.ts [hunk 3]
```diff
@@ -693,8 +694,21 @@ export class CloudApi {
...
+    sessionId,
+    userId,
+    shortId,
+  }: {
+    projectId: string
+    sessionId: string
+    userId: string
+    shortId?: string
+  }) {
+    let path = `/projects/${projectId}?sessionId=${sessionId}&userId=${userId}`
```
Comment (vvagaytsev) [inline core/src/cloud/api.ts:708]: Nit :) This can be `const` with a ternary operator. No need to create this string object if `shortId` is `true`.

Comment (shumailxyz) [reply core/src/cloud/api.ts:708]: True. I just find it bit more readable compared to ternary operator here as the string is bit long.

**Pull request merged**